src/Toodle.ts

```diff
@@ -81,62 +79,61 @@ export class Toodle {
    * see {@link Toodle.attach} for creating a Toodle instance that draws to a canvas.
    */
   constructor(
-    device: GPUDevice,
-    context: GPUCanvasContext,
-    presentationFormat: GPUTextureFormat,
+    backend: IRenderBackend,
     canvas: HTMLCanvasElement,
     resolution: Resolution,
     options: ToodleOptions,
   ) {
-    this.#limits = {
-      ...DEFAULT_LIMITS,
-      ...options.limits,
-    };
+    this.#backend = backend;
     this.#matrixPool = new Pool<Mat3>(
       () => mat3.identity(),
-      this.#limits.instanceCount,
+      backend.limits.instanceCount,
     );
-    this.#device = device;
-    this.#context = context;
-    this.#presentationFormat = presentationFormat;
     this.#defaultFilter = options.filter ?? "linear";
-    this.assets = new AssetManager(device, presentationFormat, this.#limits);
-    this.#atlasSize = {
-      width: this.assets.textureAtlas.width,
-      height: this.assets.textureAtlas.height,
-    };
-    this.debug = { device, presentationFormat };
+
+    // Create AssetManager with the backend
+    this.assets = new AssetManager(backend);
+
+    // Set debug info for WebGPU backend
+    if (backend.type === "webgpu") {
```

Comment:
yeah this is another reason to move `device` and `presentationFormat` to the webgpu backend. it's ok to make a breaking change to `debug.device`